File: wear/src/main/java/info/nightscout/androidaps/interaction/menus/FillMenuActivity.java
package info.nightscout.androidaps.interaction.menus;

import android.content.Intent;
import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

import info.nightscout.androidaps.R;
import info.nightscout.androidaps.data.ListenerService;
import info.nightscout.androidaps.interaction.actions.FillActivity;
import info.nightscout.androidaps.interaction.utils.MenuListActivity;

/**
 * Created by adrian on 09/02/17.
 */

public class FillMenuActivity extends MenuListActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTitle(R.string.menu_prime_fill);
        super.onCreate(savedInstanceState);
    }

    @Override
    protected List<MenuItem> getElements() {
        List<MenuItem> menuItems = new ArrayList<>();
        menuItems.add(new MenuItem(R.drawable.ic_canula, getString(R.string.action_preset_1)));
        menuItems.add(new MenuItem(R.drawable.ic_canula, getString(R.string.action_preset_2)));
        menuItems.add(new MenuItem(R.drawable.ic_canula, getString(R.string.action_preset_3)));
        menuItems.add(new MenuItem(R.drawable.ic_canula, getString(R.string.action_free_amount)));

        return menuItems;
    }

    @Override
    protected void doAction(String action) {
        if (getString(R.string.action_preset_1).equals(action)) {
            ListenerService.initiateAction(this, "fillpreset 1");
        } else if (getString(R.string.action_preset_2).equals(action)) {
            ListenerService.initiateAction(this, "fillpreset 2");
        } else if (getString(R.string.action_preset_3).equals(action)) {
            ListenerService.initiateAction(this, "fillpreset 3");
        } else if (getString(R.string.action_free_amount).equals(action)) {
            Intent intent = new Intent(this, FillActivity.class);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            this.startActivity(intent);
        }
    }
}

File: wear/src/main/kotlin/app/aaps/wear/interaction/menus/FillMenuActivity.kt
package app.aaps.wear.interaction.menus

import android.content.Intent
import android.os.Bundle
import app.aaps.core.interfaces.rx.events.EventWearToMobile
import app.aaps.core.interfaces.rx.weardata.EventData
import app.aaps.wear.R
import app.aaps.wear.interaction.actions.FillActivity
import app.aaps.wear.interaction.utils.MenuListActivity

class FillMenuActivity : MenuListActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        setTitle(R.string.menu_prime_fill)
        super.onCreate(savedInstanceState)
    }

    override fun provideElements(): List<MenuItem> =
        ArrayList<MenuItem>().apply {
            add(MenuItem(R.drawable.ic_canula, getString(R.string.action_preset_1)))
            add(MenuItem(R.drawable.ic_canula, getString(R.string.action_preset_2)))
            add(MenuItem(R.drawable.ic_canula, getString(R.string.action_preset_3)))
            add(MenuItem(R.drawable.ic_canula, getString(R.string.action_free_amount)))
        }

    override fun doAction(position: String) {
        when (position) {
            getString(R.string.action_preset_1)    -> rxBus.send(EventWearToMobile(EventData.ActionFillPresetPreCheck(1)))
            getString(R.string.action_preset_2)    -> rxBus.send(EventWearToMobile(EventData.ActionFillPresetPreCheck(2)))
            getString(R.string.action_preset_3)    -> rxBus.send(EventWearToMobile(EventData.ActionFillPresetPreCheck(3)))
            getString(R.string.action_free_amount) -> startActivity(Intent(this, FillActivity::class.java))
        }
    }
}
